ui/src/app/services/leaderboard.service.ts

```diff
@@ -10,8 +11,9 @@ export class LeaderboardService {
 
   constructor(
     private apiService: ApiService,
-    private userSession: UserSessionService
-  ) {}
+    private userSession: UserSessionService,
```

Comment:
You've replaced the UserSessionService with functionality in the UserService. Remove UserSession, as it's now superfluous.